packages/core/src/modules/presentation-exchange/utils/credentialSelection.ts

```diff
@@ -36,8 +37,12 @@ export async function selectCredentialsForRequest(
     ...selectResultsRaw,
     // Map the encoded credential to their respective w3c credential record
     verifiableCredential: selectResultsRaw.verifiableCredential?.map((encoded) => {
-      const credentialIndex = encodedCredentials.indexOf(encoded)
-      const credentialRecord = credentialRecords[credentialIndex]
+      const credentialRecord = credentialRecords.find((record) => {
+        const originalVc = getSphereonOriginalVerifiableCredential(record.credential)
+
+        return deepEquality(originalVc, encoded)
+      })
```

Comment:
It feels a bit expensive to do a deepEquality on every credential to find the original record that was associated with VC. Is there no other way to do the matching?

Because we have the encodedCredenitals, we can do a `===` check on it right, as the object references are the same?

So in that case `const credentialIndex = encodedCredentials.indexOf(encoded)` should still work

But maybe I'm missing something